src/søknad/steg/4-barnasbosted/BarnetsBostedEndre.tsx

```diff
@@ -217,129 +149,120 @@ const BarnetsBostedEndre: React.FC<Props> = ({
+                    : `${hentTekst('barnekort.medforelder.hemmelig', intl)}, ${
+                        barn.medforelder.verdi.alder
+                      }`}
+                </BodyShort>
               </>
             )}
+          </SeksjonGruppe>
+        )}
+
+        {visBorAnnenForelderINorge && (
+          <BorForelderINorge
+            barn={barn}
+            forelder={forelder}
+            settForelder={settForelder}
+            settDokumentasjonsbehovForBarn={settDokumentasjonsbehovForBarn}
+          />
+        )}
+
+        {(visBostedOgSamværSeksjon(forelder, visBorAnnenForelderINorge) ||
+          barnHarSammeForelder) && (
+          <BostedOgSamvær
+            settForelder={settForelder}
+            forelder={forelder}
+            barn={barn}
+            settDokumentasjonsbehovForBarn={settDokumentasjonsbehovForBarn}
+          />
+        )}
 
-          {erForelderUtfylt(forelder) &&
-            (erIdentUtfyltOgGylding(forelder.ident?.verdi) ||
-              erFødselsdatoUtfyltOgGyldigEllerTomtFelt(
-                forelder?.fødselsdato?.verdi
-              ) ||
-              utfyltNødvendigSpørsmålUtenOppgiAnnenForelder(forelder) ||
-              harForelderFraPdl) && (
-              <Button variant="secondary" onClick={leggTilForelder}>
-                <LocaleTekst
-                  tekst={
-                    !sisteBarnUtfylt && !erPåSisteBarn
-                      ? 'knapp.neste.barn'
-                      : 'knapp.neste'
-                  }
+        {!barnHarSammeForelder &&
+          visSpørsmålHvisIkkeSammeForelder(forelder) && (
+            <>
+              {forelder.borINorge?.verdi && (
+                <BorAnnenForelderISammeHus
+                  forelder={forelder}
+                  settForelder={settForelder}
+                  barn={barn}
                 />
-              </Button>
-            )}
-        </div>
+              )}
+
+              {skalFylleUtHarBoddSammenFør && (
+                <BoddSammenFør
+                  forelder={forelder}
+                  barn={barn}
+                  settForelder={settForelder}
+                />
+              )}
+              {(boddSammenFør?.svarid === ESvar.NEI ||
+                erGyldigDato(flyttetFra?.verdi)) && (
+                <HvorMyeSammen
+                  forelder={forelder}
+                  barn={barn}
+                  settForelder={settForelder}
+                />
+              )}
+            </>
+          )}
+
+        {erForelderUtfylt(forelder) &&
+          (erIdentUtfyltOgGylding(forelder.ident?.verdi) ||
+            erFødselsdatoUtfyltOgGyldigEllerTomtFelt(
+              forelder?.fødselsdato?.verdi
+            ) ||
+            utfyltNødvendigSpørsmålUtenOppgiAnnenForelder(forelder) ||
+            harForelderFraPdl) && (
+            <Button variant="secondary" onClick={leggTilForelder}>
+              <LocaleTekst tekst={'knapp.neste'} />
+            </Button>
```

Comment:
Logikk for om det skal stå "neste barn" eller "neste" er fjernet da det kun har stått "neste" på knappen de siste 3 årene.